ark/phenotyping/preprocess.py

```diff
@@ -10,6 +11,60 @@
 from ark.utils import load_utils
 from ark.utils import misc_utils
 
+from timeit import default_timer
+
+
+def _unpacking_apply_along_axis(func_args):
+    """Maps apply_along_axis to each array subset
+
+    Args:
+        func_args (tuple):
+            The arguments to pass into apply_along_axis
+
+    Returns:
+        numpy.ndarray:
+            The result for apply_along_axis for a specific array subset
+    """
+
+    (func1d, axis, arr, args, kwargs) = func_args
+    return np.apply_along_axis(func1d, axis, arr, *args, **kwargs)
+
+
+def _parallel_apply_along_axis(func1d, axis, arr, *args, **kwargs):
+    """Runs apply_along_axis in parallel for a massive speedup (but at a massive memory cost too)
+
+    Args:
+        func1d (Callable):
+            The function to apply along the given axis
+        axis (int):
+            The axis to run
+
+    Returns:
+        numpy.ndarray:
+            The processed array, equivalent to what np.apply_along_axis returns
+    """
+
+    # If axis is 0 we'll effectively be working with the 1st axis
+    effective_axis = 1 if axis == 0 else axis
+
+    # Swap the 0th and 1st axis if axis specified is 0
+    if effective_axis != axis:
+        arr = arr.swapaxes(axis, effective_axis)
+
+    # Map the array subsets
+    chunks = [(func1d, effective_axis, sub_arr, args, kwargs)
+              for sub_arr in np.array_split(arr, multiprocessing.cpu_count())]
+
+    # Run the process in parallel
+    pool = multiprocessing.Pool()
+    individual_results = pool.map(_unpacking_apply_along_axis, chunks)
+
```

Comment:
Have you looked at using `concurrent.futures` instead?  I think the chunking will be the same but it might be worth comparing the performance between a threadPool vs a processPool (plus the API is a little better)